src/adc_readout/AdcReadoutBase.cpp

```diff
@@ -125,51 +126,59 @@ void AdcReadoutBase::packetFunction(InData const &Packet,
 }
 
 void AdcReadoutBase::inputThread() {
-  std::vector<std::unique_ptr<UDPClient>> UDPReceivers;
+  std::unique_ptr<UDPClient> Receiver1;
 
   std::function<SamplingRun *(ChannelID)> DataModuleProducer(
       [this](auto Identifier) { return this->GetDataModule(Identifier); });
 
   std::function<bool(SamplingRun *)> QueingFunction1([this](SamplingRun *Data) {
-    this->AdcStats.current_ts_sec = Data->TimeStamp.Seconds;
+    this->AdcStats.current_ts_sec = Data->StartTime.getSeconds();
     return this->QueueUpDataModule(Data);
   });
-
   PacketParser Parser1(QueingFunction1, DataModuleProducer, 0);
-  UDPReceivers.emplace_back(std::make_unique<UDPClient>(
-      Service, EFUSettings.DetectorAddress, EFUSettings.DetectorPort,
-      [&Parser1, this](auto Packet) {
-        this->packetFunction(Packet, Parser1);
-      }));
-
-  std::function<bool(SamplingRun *)> QueingFunction2([this](SamplingRun *Data) {
-    this->AdcStats.current_ts_alt_sec = Data->TimeStamp.Seconds;
-    return this->QueueUpDataModule(Data);
-  });
 
-  PacketParser Parser2(QueingFunction2, DataModuleProducer, 1);
-  if (not ReadoutSettings.AltDetectorInterface.empty() and
-      ReadoutSettings.AltDetectorPort != 0) {
-    UDPReceivers.emplace_back(std::make_unique<UDPClient>(
-        Service, ReadoutSettings.AltDetectorInterface,
-        ReadoutSettings.AltDetectorPort, [&Parser2, this](auto Packet) {
-          this->packetFunction(Packet, Parser2);
-        }));
-  }
+  auto PacketHandler1 = [&Parser1, this](auto Packet) {
+    this->packetFunction(Packet, Parser1);
+  };
+
+  auto SetUpInputForProcessing = [&]() {
+    Receiver1->setPacketHandler(PacketHandler1);
+  };
+  auto GetPacketsForConfig = [&](auto Packet) {
+    try {
+      auto Config = parseHeaderForConfigInfo(Packet);
+      TimestampOffset = OffsetTime(ReadoutSettings.TimeOffsetSetting,
+                                   ReadoutSettings.ReferenceTime,
+                                   Config.BaseTime.getTimeStampNS());
+      LOG(INIT, Sev::Info, "Config packet received, starting data processing.");
+      SetUpInputForProcessing();
```

Comment:
Couldyou not just do 
Receiver1->setPacketHandler(PacketHandler1);
here?

Reply:
True, done.